### PR TITLE
Ensure we exchange cookies in s_server even if SCTP is disabled

### PR DESCRIPTION
**Pull request by mattcaswell**

### CODE REVIEW

File: apps/s_server.c [hunk 1]
```diff
@@ -2262,11 +2262,10 @@ static int sv_body(int s, int stype, int prot, unsigned char *context)
             BIO_ctrl(sbio, BIO_CTRL_DGRAM_MTU_DISCOVER, 0, NULL);
```
Comment (dot-asm) [inline apps/s_server.c]: While you are at it, it might be appropriate to fix "if-if-has-{}-then-else-has-to-have-pair-too" styling rule.

Comment (dot-asm) [inline apps/s_server.c]: Last remark was a *side note*! I.e. not replying is a viable option.

File: apps/s_server.c [hunk 2]
```diff
@@ -2262,11 +2262,10 @@ static int sv_body(int s, int stype, int prot, unsigned char *context)
...
 
 # ifndef OPENSSL_NO_SCTP
```
Comment (dot-asm) [inline apps/s_server.c]: On another *side note* why not #ifdef IPPROTO_SCTP?

Comment (mattcaswell) [reply apps/s_server.c]: `prot` will never be set to `IPPROTO_SCTP` if OPENSSL_NO_SCTP is defined

**Comment by mattcaswell** [comment]
Pushed. Thanks.

File: apps/s_server.c (hunk 2 above)
Comment (dot-asm) [reply apps/s_server.c]: > prot will never be set to IPPROTO_SCTP if OPENSSL_NO_SCTP is defined

Exactly!

Comment (mattcaswell) [reply apps/s_server.c]: So, in my mind that makes it better to check `OPENSSL_NO_SCTP`.

**Pull request closed without merging**